builtin/providers/aws/resource_aws_route.go

```diff
@@ -0,0 +1,337 @@
+	d.Set("gateway_id", route.GatewayId)
+	d.Set("instance_id", route.InstanceId)
+	d.Set("instance_owner_id", route.InstanceOwnerId)
+	d.Set("network_interface_id", route.NetworkInterfaceId)
+	d.Set("origin", route.Origin)
+	d.Set("state", route.State)
+	d.Set("vpc_peering_connection_id", route.VpcPeeringConnectionId)
+
+	return nil
+}
+
+func resourceAwsRouteUpdate(d *schema.ResourceData, meta interface{}) error {
+	if d.HasChange("destination_cidr_block") {
+		return resourceAwsRouteRecreate(d, meta)
+	}
+
+	conn := meta.(*AWSClient).ec2conn
+	var numTargets int
+	var setTarget string
+	allowedTargets := []string{
+		"gateway_id",
+		"instance_id",
+		"network_interface_id",
+		"vpc_peering_connection_id",
+	}
+	replaceOpts := &ec2.ReplaceRouteInput{}
+
+	// Check if more than 1 target is specified
+	for _, target := range allowedTargets {
+		if len(d.Get(target).(string)) > 0 {
+			numTargets++
+			setTarget = target
+		}
+	}
+
+	if numTargets > 1 {
+		fmt.Errorf("Error: more than 1 target specified. Only 1 of gateway_id" +
+			"instance_id, network_interface_id, route_table_id or" +
+			"vpc_peering_connection_id is allowed.")
+	}
+
+	// Formulate ReplaceRouteInput based on the target type
+	switch setTarget {
+	case "gateway_id":
+		replaceOpts = &ec2.ReplaceRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			GatewayId:            aws.String(d.Get("gateway_id").(string)),
+		}
+	case "instance_id":
+		replaceOpts = &ec2.ReplaceRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			InstanceId:           aws.String(d.Get("instance_id").(string)),
+			//NOOP: Ensure we don't blow away network interface id that is set after instance is launched
+			NetworkInterfaceId:	  aws.String(d.Get("network_interface_id").(string)),
+		}
+	case "network_interface_id":
+		replaceOpts = &ec2.ReplaceRouteInput{
+			RouteTableId:         aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock: aws.String(d.Get("destination_cidr_block").(string)),
+			NetworkInterfaceId:   aws.String(d.Get("network_interface_id").(string)),
+		}
+	case "vpc_peering_connection_id":
+		replaceOpts = &ec2.ReplaceRouteInput{
+			RouteTableId:           aws.String(d.Get("route_table_id").(string)),
+			DestinationCidrBlock:   aws.String(d.Get("destination_cidr_block").(string)),
+			VpcPeeringConnectionId: aws.String(d.Get("vpc_peering_connection_id").(string)),
+		}
+	default:
+		fmt.Errorf("Error: invalid target type specified.")
+	}
+	log.Printf("[DEBUG] Route replace config: %s", awsutil.Prettify(replaceOpts))
+
+	// Replace the route
+	_, err := conn.ReplaceRoute(replaceOpts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsRouteRecreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Is there a specific reason to use this method and strategy for recreating the route with a change of `destination_cidr_block`, vs. simply marking the `destination_cidr_block` as `ForceNew: true` in the schema and let Terraform do the recreation?